File: app/src/main/java/kr/co/company/capstone/dto/ErrorMessage.java
package kr.co.company.capstone.dto;

import com.google.gson.Gson;

import lombok.Data;
import retrofit2.Response;

@Data
public class ErrorMessage {
    private String timestamp;
    private int status;
    private String error;
    private String code;
    private String message;

    public static ErrorMessage getErrorByResponse(Response response) {
        return new Gson().fromJson(response.errorBody().charStream(), ErrorMessage.class);
    }
}

File: app/src/main/java/kr/co/company/capstone/dto/ErrorMessage.kt
package kr.co.company.capstone.dto

import com.google.gson.Gson
import retrofit2.Response

data class ErrorMessage(
    val timestamp: String? = null,
    val status: Int = 0,
    val error: String? = null,
    val code: String? = null,
    val message: String? = null
) {
    companion object {
        fun getErrorByResponse(response: Response<*>): ErrorMessage {
            response.errorBody()?.charStream()?.let {
                return Gson().fromJson(it, ErrorMessage::class.java)
            }
            throw IllegalArgumentException("Invalid Response error")
        }
    }
}